game/world/objects/interactive.cpp

```diff
@@ -291,22 +306,49 @@ void Interactive::implTick(Pos& p, uint64_t /*dt*/) {
   loopState = (prev==state);
   }
 
-void Interactive::nextState(Npc& npc) {
-  const int prev = state;
-  if(!setAnim(&npc,Anim::In))
-    return;
-  setState(std::min(stateNum,state+1));
-  loopState = (prev==state);
+void Interactive::nextState(Npc& npc, const uint8_t st) {
+  wait = !st;
+  switch(st){
```

Comment:
`enum` ?

Reply:
I was reluctant because i thought it's unnecessary to have it in class definition only for ladder. But since you said `nextState` could be used for regular Mobsis in mods as well it' makes sense. And not as ugly as raw numbers of course.